### PR TITLE
libbpf-tools: Allow tools to run on old kernels

### PR DESCRIPTION
**Pull request by michael-chuh**
Add more tools to fallback to use kprobes if kernel lacks support for fentry/fexit,
discussion is on https://github.com/iovisor/bcc/issues/4231

### CODE REVIEW

File: libbpf-tools/biostacks.bpf.c [hunk 1]
```diff
@@ -67,20 +67,8 @@ int trace_start(void *ctx, struct request *rq, bool merge_bio)
...
-SEC("kprobe/blk_account_io_merge_bio")
-int BPF_KPROBE(blk_account_io_merge_bio, struct request *rq)
-{
-	return trace_start(ctx, rq, true);
-}
-
-SEC("fentry/blk_account_io_done")
-int BPF_PROG(blk_account_io_done, struct request *rq)
+static __always_inline
+int trace_done(void *ctx, struct request *rq)
```
Comment (chenhengqi) [inline libbpf-tools/biostacks.bpf.c:71]: Keep this in single line. We don't need `__always_inline` here.

File: libbpf-tools/readahead.bpf.c [hunk 2]
```diff
@@ -24,18 +24,8 @@ struct {
...
-	u32 pid = bpf_get_current_pid_tgid();
-	u64 one = 1;
-
-	bpf_map_update_elem(&in_readahead, &pid, &one, 0);
-	return 0;
-}
-
-SEC("fexit/__page_cache_alloc")
-int BPF_PROG(page_cache_alloc_ret, gfp_t gfp, struct page *ret)
+static __always_inline
```
Comment (chenhengqi) [inline libbpf-tools/readahead.bpf.c:27]: Ditto.

**Review by chenhengqi — COMMENTED** [review]
Please separate the changes per tool.

**Comment by michael-chuh** [comment]
I have separated it into 3 patches, and remove '__always_inline '
https://github.com/iovisor/bcc/pull/4549
https://github.com/iovisor/bcc/pull/4550
https://github.com/iovisor/bcc/pull/4551

**Comment by chenhengqi** [comment]
Close this in favor of new ones.

**Pull request closed without merging**